q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,311 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from q2_metadata.normalization._norm_errors import (
+    RuleError, ExpectedError, OntologyError, RemapError,
+    ValidationError, NormalizationError,
+    BlankError, MissingError, FormatError
+)
+
+
+def check_rule(variable: str, rules: dict, rule: str, rule_value):
+    """For each rule in the current variable's rules set,
+    perform checks that the rule values to be used for the rules
+    to apply are properly formatted.
+        - If properly formatted: full the rules dict
+        for the variable's Rules() class instance.
+        - If not properly formatted: raise a specific error and
+        collect the encountered error in the ErrorsCollection()
+        instance for this variable's Rules() class instance.
```

Comment:
This sort of information would usually be documented under a `Raises` section. Check the docstring reference:

https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard

Reply:
Of course! 
Will do.

Reply:
And as a follow up to the above discussion, I'm not sure for the "Raises" part:

- shall I make an entry per error (current solution)? 

```
    Raises
    ------
    RuleError
        If the rule is not recognized.
    ExpectedError
        If the "expected" rule if not properly formatted.
    OntologyError
        If the "ontology" rule if not properly formatted.
    RemapError
        If the "remap" rule if not properly formatted.
    ValidationError
        If the "validation" rule if not properly formatted.
    NormalizationError
        If the "normalization" rule if not properly formatted.
    BlankError
        If the "blank" rule if not properly formatted.
    MissingError
        If the "missing" rule if not properly formatted.
    FormatError
        If the "format" rule if not properly formatted.
```

- or just this?

```
    Raises
    ------
    User-defined error specific to each rule. Can be one of
    RuleError, ExpectedError, OntologyError, RemapError,
    ValidationError, NormalizationError, BlankError,
    MissingError, FormatError.
```

Reply:
the raised errors are removed